src/pudl/analysis/record_linkage/model_helpers.py

```diff
@@ -1,18 +1,134 @@
 """This module provides shared tooling that can be used by all record linkage models."""
 import importlib
+from collections.abc import Callable
 
+import mlflow
 import yaml
+from dagster import Config, op
+from pydantic import BaseModel
 
+import pudl
+from pudl.workspace.setup import PudlPaths
 
-def get_model_config(model_key: str) -> dict:
+logger = pudl.logging_helpers.get_logger(__name__)
+
+
+def get_model_config(experiment_name: str) -> dict:
     """Load model configuration from yaml file."""
     config_file = (
         importlib.resources.files("pudl.package_data.settings")
         / "record_linkage_model_config.yml"
     )
     config = yaml.safe_load(config_file.open("r"))
 
-    if not (model_config := config.get(model_key)):
-        raise RuntimeError(f"No {model_key} entry in {config_file}")
+    if not (model_config := config.get(experiment_name)):
+        raise RuntimeError(f"No {experiment_name} entry in {config_file}")
+
+    return {experiment_name: model_config}
+
+
+def flatten_model_config(model_config: dict) -> dict:
```

Comment:
This converts a nested dictionary of configuration and turns it into a flat dictionary which works better in the `mlflow` UI. For example:

```py
config = {
    "ferc_to_ferc": {
        "link_ids_cross_year": {
            "compute_distance_matrix": {"distance_threshold": .5, "metric": "euclidean"},
            "match_orphaned_records": {"distance_threshold": 0.5},
            ...
        {
    }
}
```

becomes:
```py
config = {
    "ferc_to_ferc.link_ids_cross_year.compute_distance_matrix.distance_threshold": .5,
    "ferc_to_ferc.link_ids_cross_year.compute_distance_matrix.metric": "euclidean",
    "ferc_to_ferc.match_orphaned_records.compute_distance_matrix.distance_threshold": .5,
    ...
}